softvis3d-frontend/src/services/AutoReloadService.ts

```diff
@@ -0,0 +1,36 @@
+import { AppStatusStore } from "../stores/AppStatusStore";
+import SonarQubeComponentInfoService from "./sonarqube/SonarQubeComponentInfoService";
+import { isUndefined } from "util";
+
+export default class AutoReloadService {
+
+    // 5 minutes
+    public static RELOAD_INTERVAL_MS = 5 * 60 * 1000;
+
+    private appStatusStore: AppStatusStore;
+    private componentInfoService: SonarQubeComponentInfoService;
+    private timer: number | undefined;
+
+    constructor(appStatusStore: AppStatusStore, componentInfoService: SonarQubeComponentInfoService) {
+        this.appStatusStore = appStatusStore;
+        this.componentInfoService = componentInfoService;
+    }
+
+    public startAutoReload(): void {
+        if (this.timer) {
+            window.clearInterval(this.timer);
+        }
+
+        this.timer = window.setInterval(this.loadComponentInfo, AutoReloadService.RELOAD_INTERVAL_MS);
```

Comment:
When testing the feature I had problems with the unbound "this", at execution time.

Either of those would take care of the problem:
```js
window.setInterval(() => this.loadComponentInfo(), AutoReloadService.RELOAD_INTERVAL_MS)
window.setInterval(this.loadComponentInfo.bind(this), AutoReloadService.RELOAD_INTERVAL_MS)
```

Reply:
ups. thanks. is fixed now.